app/controllers/vm_infra_controller.rb

```diff
@@ -73,4 +73,16 @@ def skip_breadcrumb?
 
   menu_section :inf
   has_custom_buttons
+
+  def vm_transform
+    vm = Vm.find_by(:id => params[:id].to_i)
+    @right_cell_text = _("Transform VM %{name} to RHV") % {:name => vm.name}
+    dialog = Dialog.find_by(:label => 'Transform VM')
```

Comment:
OK, so, this is evil, but there's no other way, not for fine. So, I'd suggest doing a post-fine PR that adds some kind of `identifier` field and uses that instead. (And possibly consider hiding these magical dialogs from the UI.)